quickstart/backend/src/main/java/com/digitalasset/quickstart/service/LicenseApiImpl.java

```diff
@@ -55,135 +65,197 @@ public LicenseApiImpl(
+                                LoggingSpanHelper.recordException(methodSpan, ex);
+                            }
+                        })
+                );
     }
 
     @Override
+    @WithSpan
     public CompletableFuture<ResponseEntity<Void>> renewLicense(
-            String contractId,
-            String commandId,
-            org.openapitools.model.LicenseRenewRequest licenseRenewRequest
+            @SpanAttribute("license.contractId") String contractId,
+            @SpanAttribute("license.commandId") String commandId,
+            LicenseRenewRequest licenseRenewRequest
     ) {
-        logger.info("Renewing License with contractId: {}", contractId);
-        String providerParty = authenticatedPartyService.getPartyOrFail();
-
-        return damlRepository.findLicenseById(contractId).thenCompose(contract -> {
-            Duration extDuration = Duration.parse(licenseRenewRequest.getLicenseExtensionDuration());
-            long extensionMicros = extDuration.toNanos() / 1000;
-            RelTime licenseExtensionDuration = new RelTime(extensionMicros);
-
-            Duration payDuration = Duration.parse(licenseRenewRequest.getPaymentAcceptanceDuration());
-            long payDurationMicros = payDuration.toNanos() / 1000;
-            RelTime paymentAcceptanceDuration = new RelTime(payDurationMicros);
-
-            String description = licenseRenewRequest.getDescription();
-
-            License_Renew choice = new License_Renew(
-                    licenseRenewRequest.getLicenseFeeCc(),
-                    licenseExtensionDuration,
-                    paymentAcceptanceDuration,
-                    description
-            );
-
-            return ledger.exerciseAndGetResult(providerParty, contract.contractId, choice, commandId)
-                    .thenApply((Tuple2<ContractId<LicenseRenewalRequest>, ContractId<AppPaymentRequest>> result) -> {
-                        logger.info("Successfully renewed License {}. RenewalRequest ContractId: {} and PaymentRequest ContractId: {}",
-                                contractId, result.get_1.getContractId, result.get_2.getContractId);
-                        return ResponseEntity.ok().<Void>build();
-                    }).exceptionally(ex -> {
-                        logger.error("Error renewing License", ex);
-                        return ResponseEntity.status(HttpStatus.INTERNAL_SERVER_ERROR).build();
-                    });
-        }).exceptionally(ex -> {
-            logger.error("Error fetching License with contractId: {}", contractId, ex);
-            return ResponseEntity.status(HttpStatus.INTERNAL_SERVER_ERROR).build();
-        });
+        Span methodSpan = Span.current();
+        Context parentContext = Context.current();
+
+        Map<String, Object> attributes = Map.of(
+                "license.contractId", contractId,
+                "license.commandId", commandId,
```

Comment:
These match the span (good), but don't match the use of commandId elsewhere (bad).